pkg/repository/base.go

```diff
@@ -40,10 +40,15 @@ type ContainerRepository interface {
 	UpdateContainerStatus(string, types.ContainerStatus, time.Duration) error
 	DeleteContainerState(*types.ContainerRequest) error
 	SetWorkerAddress(containerId string, addr string) error
+	SetContainerStateWithConcurrencyLimit(quota *types.ConcurrencyLimit, request *types.ContainerRequest) error
 	GetWorkerAddress(containerId string) (string, error)
 	GetActiveContainersByStubId(stubId string) ([]types.ContainerState, error)
 	GetActiveContainersByWorkspaceId(workspaceId string) ([]types.ContainerState, error)
 	GetFailedContainerCountByStubId(stubId string) (int, error)
+
+	// Caching concurrency limit TODO: discuss if this should be here
```

Comment:
@luke-lombardi This is for the caching of the concurrency limit in redis

Reply:
@jsun-m Should we do another PR after this is merged which moves the SetConcurrencyLimitByWorkspaceId over to a Workspace Repo?